nvmeadm/src/nvmf_discovery.rs

```diff
@@ -403,3 +405,23 @@ impl DiscoveryLogEntry {
         Ok(connect_args)
     }
 }
+/// This method disconnects a specific NVMf device, identified by its nqn.
+///
+///  # Example
+///  ```rust
+///  let num_disconnects = nvmeadm::nvmf_discovery::disconnect(nqn);
+///  ```
+
+pub fn disconnect(nqn: &str) -> Result<u32, Error> {
+    let mut disconnected_devices: u32 = 0;
+    let _subsys: Result<Vec<Subsystem>, Error> = NvmeSubsystems::new()?
+        .filter_map(Result::ok)
+        .filter(|e| e.nqn == nqn)
+        .map(|e| {
+            disconnected_devices += 1;
+            e.disconnect()?;
+            Ok(e)
+        })
+        .collect();
+    Ok(disconnected_devices)
```

Comment:
I think you can return subsys.unwrap().len() instead of maintaining an additional counter (and the return type has to change to usize).